src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -703,58 +705,31 @@ private BoundStatement BindDeclarationStatementParts(LocalDeclarationStatementSy
-            MethodSymbol disposeMethod = FindPatternMethodRelaxed(typeExpr.Type,
-                                                                  WellKnownMemberNames.DisposeMethodName,
-                                                                  lookupResult,
-                                                                  syntaxNode,
-                                                                  warningsOnly: true,
-                                                                  initialDiagnostics,
-                                                                  syntaxNode.SyntaxTree,
-                                                                  MessageID.IDS_Disposable);
-
-            // we failed if we didn't find a void returning method; try searching extension methods
-            // PROTOTYPE: what happens if there is a no arg int returning method, but a params object[] void returning. Currently, overload resolution would choose the int return, and we'd fail here.
-            if (disposeMethod is null || !disposeMethod.ReturnsVoid)
-            {
-                lookupResult.Clear();
-
-                var extDisposeMethod = FindPatternExtensionMethod(typeExpr,
-                                                                  WellKnownMemberNames.DisposeMethodName,
-                                                                  syntaxNode,
-                                                                  diagnostics,
-                                                                  MessageID.IDS_Disposable);
-
-                // if we still can't satisfy the lookup via extension method, report the diagnostics from both lookups
-                if (extDisposeMethod is null || !extDisposeMethod.ReturnsVoid)
-                {
-                    diagnostics.AddRange(initialDiagnostics);
-                    if (disposeMethod?.ReturnsVoid == false)
-                    {
-                        ReportPatternWarning(diagnostics, typeExpr.Type, disposeMethod, syntaxNode, MessageID.IDS_Disposable);
-                    }
-                    if (extDisposeMethod?.ReturnsVoid == false)
-                    {
-                        ReportPatternWarning(diagnostics, typeExpr.Type, extDisposeMethod, syntaxNode, MessageID.IDS_Disposable);
-                    }
-                    extDisposeMethod = null;
-                }
+        /// <param name="expr">Expression on which to perform lookup</param>
+        /// <param name="syntaxNode">The syntax node to perform lookup on</param>
+        /// <param name="diagnostics">Populated with invocation errors, and warnings of near misses</param>
+        /// <returns>The <see cref="MethodSymbol"/> of the Dispose method if one is found, otherwise null.</returns>
+        internal MethodSymbol TryFindDisposePatternMethod(BoundExpression expr, SyntaxNode syntaxNode, DiagnosticBag diagnostics)
+        {
+            if(expr?.Type is null)
```

Comment:
>expr [](start = 15, length = 4)

Is `expr` ever `null`?

Reply:
It shouldn't be. I've added an assert to declare the precondition.